chapters/solutions.Rmd

```diff
@@ -868,11 +868,77 @@ FIXME
 
 ## Chapter \@ref(config)
 
+### Exercise \@ref(config-ex-build-plotparams)
+
+The build rule involving `plotcounts.py` should now
+read as follows:
+
+```make
+## results/collated.png: plot the collated results.
+results/collated.png : results/collated.csv bin/plotparams.yml
+	python $(PLOT) $< --outfile $@ --plotparams $(word 2,$^)
+```
+
+For reference, the complete `Makefile` is now:
+```make
+.PHONY: all clean help settings
+
+COUNT=bin/countwords.py
+COLLATE=bin/collate.py
+PLOT=bin/plotcounts.py
+DATA=$(wildcard data/*.txt)
+RESULTS=$(patsubst data/%.txt,results/%.csv,$(DATA))
+
+## all : regenerate all results.
+all : results/collated.png
+
+## results/collated.png: plot the collated results.
+results/collated.png : results/collated.csv bin/plotparams.yml
+	python $(PLOT) $< --outfile $@ --plotparams $(word 2,$^)
+
+## results/collated.csv : collate all results.
+results/collated.csv : $(RESULTS) $(COLLATE)
+	@mkdir -p results
+	python $(COLLATE) $(RESULTS) > $@
+
+## results/%.csv : regenerate result for any book.
+results/%.csv : data/%.txt $(COUNT)
+	python $(COUNT) $< > $@
+
+## clean : remove all generated files.
+clean :
+	rm $(RESULTS) results/collated.csv results/collated.png
+
+## settings : show variables' values.
+settings :
+	@echo COUNT: $(COUNT)
+	@echo DATA: $(DATA)
+	@echo RESULTS: $(RESULTS)
+	@echo COLLATE: $(COLLATE)
+	@echo PLOT: $(PLOT)
+
+## help : show this message.
+help :
+	@grep '^##' ./Makefile
+```
+
 ### Exercise \@ref(config-ex-accessible)
 
 FIXME
 
-### Exercise \@ref(config-ex-debug)
+### Exercise \@ref(config-ex-saveload)
+
+FIXME
```

Comment:
Would you be able to add the solution here? It needs to build on the repo version of [`plotcounts.py`](https://github.com/merely-useful/merely-useful.github.io/blob/book/zipf/bin/plotcounts.py`).